In [1]:
import os
from pathlib import Path

import pandas as pd

In [2]:
# Caminho absoluto até o diretório do arquivo atual
current_path = Path("__file__").resolve()
project_root = current_path.parent.parent.parent.parent
file = os.path.join(project_root, "data/srag_csvs", "INFLUD19-26-06-2025.csv")

In [3]:
df = pd.read_csv(file, sep=";")
df.head()

C:\Users\aurelio.silva_indici\AppData\Local\Temp\ipykernel_10488\984421930.py:1: DtypeWarning: Columns (92,94,115,117,118,119,121,122,129,130,144,147,150,155,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(file, sep=";")


NU_NOTIFIC  DT_NOTIFIC  SEM_NOT  DT_SIN_PRI  SEM_PRI SG_UF_NOT  \
0  315478195042  2019-01-10        2  2019-01-06        2        MG   
1  315478195276  2019-01-03        1  2019-01-01        1        SP   
2  315478207219  2019-01-02        1  2018-12-31        1        PE   
3  315478211086  2019-01-10        2  2019-01-07        2        SP   
4  315478212765  2019-01-11        2  2019-01-06        2        PE   

          ID_REGIONA  CO_REGIONA      ID_MUNICIP  CO_MUN_NOT  ... VG_OMS  \
0     BELO HORIZONTE      1449.0  BELO HORIZONTE    310620.0  ...    NaN   
1      GVE I CAPITAL      1331.0       SAO PAULO    355030.0  ...    NaN   
2                001      1497.0          RECIFE    261160.0  ...    NaN   
3  GVE XVII CAMPINAS      1342.0        CAMPINAS    350950.0  ...    NaN   
4                004      1499.0         CARUARU    260410.0  ...    NaN   

  VG_OMSOUT  VG_LIN  VG_MET  VG_METOUT  VG_DTRES  VG_ENC VG_REINF  VG_CODEST  \
0       NaN     NaN     NaN        NaN       NaN     NaN      NaN        NaN   
1       NaN     NaN     NaN        NaN       NaN     NaN      NaN        NaN   
2       NaN     NaN     NaN        NaN       NaN     NaN      NaN        NaN   
3       NaN     NaN     NaN        NaN       NaN     NaN      NaN        NaN   
4       NaN     NaN     NaN        NaN       NaN     NaN      NaN        NaN   

  REINF  
0   NaN  
1   NaN  
2   NaN  
3   NaN  
4   NaN  

[5 rows x 194 columns]

In [4]:
df.shape

(48961, 194)

### Analise de colunas

#### 0 - 10

| # | Column     | Non-Null Count | Dtype   |
| - | ---------- | -------------- | ------- |
| 0 | NU_NOTIFIC | 48961 non-null | int64   |
| 1 | DT_NOTIFIC | 48961 non-null | object  |
| 2 | SEM_NOT    | 48961 non-null | int64   |
| 3 | DT_SIN_PRI | 48961 non-null | object  |
| 4 | SEM_PRI    | 48961 non-null | int64   |
| 5 | SG_UF_NOT  | 48919 non-null | object  |
| 6 | ID_REGIONA | 44025 non-null | object  |
| 7 | CO_REGIONA | 44025 non-null | float64 |
| 8 | ID_MUNICIP | 48919 non-null | object  |
| 9 | CO_MUN_NOT | 48919 non-null | float64 |

In [17]:
df.iloc[:,0:10].info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48961 entries, 0 to 48960
Data columns (total 10 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   NU_NOTIFIC  48961 non-null  int64  
 1   DT_NOTIFIC  48961 non-null  object 
 2   SEM_NOT     48961 non-null  int64  
 3   DT_SIN_PRI  48961 non-null  object 
 4   SEM_PRI     48961 non-null  int64  
 5   SG_UF_NOT   48919 non-null  object 
 6   ID_REGIONA  44025 non-null  object 
 7   CO_REGIONA  44025 non-null  float64
 8   ID_MUNICIP  48919 non-null  object 
 9   CO_MUN_NOT  48919 non-null  float64
dtypes: float64(2), int64(3), object(5)
memory usage: 3.7+ MB


#### 10-20

| # | Column     | Non-Null Count | Dtype   |
| - | ---------- | -------------- | ------- |
| 0 | CS_SEXO    | 48961 non-null | object  |
| 1 | DT_NASC    | 48797 non-null | object  |
| 2 | NU_IDADE_N | 48961 non-null | int64   |
| 3 | TP_IDADE   | 48961 non-null | int64   |
| 4 | COD_IDADE  | 48961 non-null | int64   |
| 5 | CS_GESTANT | 48961 non-null | int64   |
| 6 | CS_RACA    | 47130 non-null | float64 |
| 7 | CS_ETINIA  | 287 non-null   | object  |
| 8 | CS_ESCOL_N | 45153 non-null | float64 |
| 9 | ID_PAIS    | 48957 non-null | object  |


In [20]:
df.iloc[:,10:20].info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48961 entries, 0 to 48960
Data columns (total 10 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   CS_SEXO     48961 non-null  object 
 1   DT_NASC     48797 non-null  object 
 2   NU_IDADE_N  48961 non-null  int64  
 3   TP_IDADE    48961 non-null  int64  
 4   COD_IDADE   48961 non-null  int64  
 5   CS_GESTANT  48961 non-null  int64  
 6   CS_RACA     47130 non-null  float64
 7   CS_ETINIA   287 non-null    object 
 8   CS_ESCOL_N  45153 non-null  float64
 9   ID_PAIS     48957 non-null  object 
dtypes: float64(2), int64(4), object(4)
memory usage: 3.7+ MB


#### 20-30

In [25]:
df.iloc[:,20:30].info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48961 entries, 0 to 48960
Data columns (total 10 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   CO_PAIS     48957 non-null  float64
 1   SG_UF       48924 non-null  object 
 2   ID_RG_RESI  44623 non-null  object 
 3   CO_RG_RESI  44623 non-null  float64
 4   ID_MN_RESI  48924 non-null  object 
 5   CO_MUN_RES  48924 non-null  float64
 6   CS_ZONA     47713 non-null  float64
 7   NOSOCOMIAL  46996 non-null  float64
 8   AVE_SUINO   47204 non-null  float64
 9   FEBRE       48740 non-null  float64
dtypes: float64(7), object(3)
memory usage: 3.7+ MB


# End